src/components/ShipFit/Slot.tsx

```diff
@@ -148,6 +149,54 @@ export const Slot = (props: { type: string; index: number; fittable: boolean; ma
     [shipSnapshot, esiItem],
   );
 
+  const onDragStart = React.useCallback(
+    (e: React.DragEvent<HTMLDivElement>) => {
+      if (!esiItem) return;
```

Comment:
Please use `esiItem === undefined`  where possible. `!esiItem` can trigger on unexpected values, making debugging harder than it needs to.